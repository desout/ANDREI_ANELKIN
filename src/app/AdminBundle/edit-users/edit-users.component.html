<div class="container">
  <app-user-list></app-user-list>
  <div class="buttons">
    <button (click)="onClickNew()" class="btn btn-success">{{'INFO.NEWBUTTON' | translate}}</button>
    <button (click)="onClickEdit()" class="btn btn-info">{{'INFO.EDITBUTTON' | translate}}</button>
  </div>
  <app-edit-user *ngIf="showEdit"></app-edit-user>
</div>
