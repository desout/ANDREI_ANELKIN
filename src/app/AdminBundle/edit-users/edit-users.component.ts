import { Component, OnInit } from '@angular/core';
import {UserDropDownService} from '../../user-dropdown.service';

@Component({
  selector: 'app-edit-users',
  templateUrl: './edit-users.component.html',
  styleUrls: ['./edit-users.component.scss']
})
export class EditUsersComponent implements OnInit {

  constructor(private dropdownService: UserDropDownService) { }
  public showEdit = false;
  ngOnInit() {
  }
  onClickEdit() {
    this.showEdit = true;
  }
  onClickNew() {
    this.dropdownService.selectedUserHandle.next(-1);
    this.showEdit = true;
  }
}
