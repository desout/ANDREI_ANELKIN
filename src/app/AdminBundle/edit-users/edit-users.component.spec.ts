import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { EditUsersComponent } from './edit-users.component';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import {TestModule} from '../../Test.module';

describe('EditUsersComponent', () => {
  let component: EditUsersComponent;
  let fixture: ComponentFixture<EditUsersComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [],
      schemas: [NO_ERRORS_SCHEMA],
      imports: [TestModule],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(EditUsersComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it( 'Button success should be created', () => {
    const element: HTMLButtonElement = fixture.nativeElement.querySelector('.btn-success');

    expect(element).toMatchSnapshot();
  });
});
