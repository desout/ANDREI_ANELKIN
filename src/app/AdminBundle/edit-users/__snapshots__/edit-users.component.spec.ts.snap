// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUsersComponent Button success should be created 1`] = `
<button
  class="btn btn-success"
>
  INFO.NEWBUTTON
</button>
`;

exports[`EditUsersComponent should create 1`] = `
<div
  id="root0"
  ng-version="7.0.3"
>
  <div
    class="container"
  >
    <app-user-list>
      <div
        class="container"
      >
        <div
          class="card"
        >
          <div>
            <app-user-list-toogle>
              <div
                class="userListToggleContainer"
              >
                <div
                  class="card-header"
                >
                  <span>
                    <div>
                      <app-user-item>
                        <div
                          class="item"
                        >
                          <div
                            class="leftSideInfo"
                          >
                            <span>
                              
                            </span>
                            <br />
                            <span>
                              
                            </span>
                            <br />
                            <span>
                              
                            </span>
                          </div>
                          <div
                            class="rightSideInfo"
                          >
                            <span>
                              
                            </span>
                          </div>
                        </div>
                      </app-user-item>
                    </div>
                    <div>
                      <app-user-list-toogle-button>
                        <div
                          class="btn btn-secondary dropdown-toggle"
                        />
                      </app-user-list-toogle-button>
                    </div>
                  </span>
                </div>
              </div>
            </app-user-list-toogle>
          </div>
          <div
            class="listView"
          >
            
          </div>
        </div>
      </div>
    </app-user-list>
    <div
      class="buttons"
    >
      <button
        class="btn btn-success"
      >
        INFO.NEWBUTTON
      </button>
      <button
        class="btn btn-info"
      >
        INFO.EDITBUTTON
      </button>
    </div>
    
  </div>
</div>
`;
